### PR TITLE
DMS - basic task replication methods implementation

### PR DESCRIPTION
**Pull request by Visya**
Adds basic implementation of DMS task replication methods.

### CODE REVIEW

**Comment by Visya** [comment]
Fixed flake8 errors.

File: tests/test_dms/test_dms.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    client.create_replication_task(
+        ReplicationTaskIdentifier="test",
+        SourceEndpointArn="source-endpoint-arn",
+        TargetEndpointArn="target-endpoint-arn",
+        ReplicationInstanceArn="replication-instance-arn",
+        MigrationType="full-load",
+        TableMappings='{"rules":[]}',
+    )
+
+    try:
```
Comment (bblommers) [inline tests/test_dms/test_dms.py:49]: Please use the `with pytest.raises(..)` construct here - that way we don't have to have an else-statement
(Example: https://github.com/spulec/moto/blob/master/tests/test_cognitoidp/test_cognitoidp.py#L1555)

File: tests/test_dms/test_dms.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        ReplicationTaskIdentifier="test",
+        SourceEndpointArn="source-endpoint-arn",
+        TargetEndpointArn="target-endpoint-arn",
+        ReplicationInstanceArn="replication-instance-arn",
+        MigrationType="full-load",
+        TableMappings='{"rules":[]}',
+    )
+    task_arn = response["ReplicationTask"]["ReplicationTaskArn"]
+
+    try:
```
Comment (bblommers) [inline tests/test_dms/test_dms.py:104]: Same here for `pytest.raises`

File: moto/dms/models.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    def start_replication_task(
+        self,
+        replication_task_arn,
+        start_replication_task_type,
+        cdc_start_time,
+        cdc_start_position,
+        cdc_stop_position,
+    ):
+        if not self.replication_tasks.get(replication_task_arn):
+            raise ResourceNotFoundFault("Replication task could not be found.")
```
Comment (bblommers) [inline moto/dms/models.py:74]: Can we get an additional test for this scenario as well

File: moto/dms/models.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+
+        return self.replication_tasks[replication_task_arn].start()
+
+    def stop_replication_task(self, replication_task_arn):
+        if not self.replication_tasks.get(replication_task_arn):
+            raise ResourceNotFoundFault("Replication task could not be found.")
```
Comment (bblommers) [inline moto/dms/models.py:80]: Same

File: moto/dms/models.py [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+
+        return self.replication_tasks[replication_task_arn].stop()
+
+    def delete_replication_task(self, replication_task_arn):
+        if not self.replication_tasks.get(replication_task_arn):
+            raise ResourceNotFoundFault("Replication task could not be found.")
```
Comment (bblommers) [inline moto/dms/models.py:86]: Same

File: moto/dms/models.py [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+
+class DatabaseMigrationServiceBackend(BaseBackend):
+    def __init__(self, region_name=None):
+        super(DatabaseMigrationServiceBackend, self).__init__()
+        self.region_name = region_name
+        self.replication_tasks = {}
+
+    def reset(self):
+        region_name = self.region_name
+        self.replication_tasks = {}
```
Comment (bblommers) [inline moto/dms/models.py:25]: This shouldn't be necessary - resetting with `self.__dict__ = {}` already wipes all data

**Review by bblommers — CHANGES_REQUESTED** [review]
Hi @Visya, thanks for your PR! Please see my comments

**Comment by Visya** [comment]
Hey @bblommers! Thank you for your review. I have fixed your comments.

**Review by bblommers — APPROVED** [review]
LGTM! Thanks for making the changes @Visya

**Pull request merged**